### PR TITLE
XW-102 Replacing AddPhotoView with ArticleAddPhotoComponent

### PR DESCRIPTION
**Pull request by bkoval**
https://wikia-inc.atlassian.net/browse/XW-102

For now it's just:

1) Replacing AddPhotoView with ArticleAddPhotoComponent
2) Renaming route, controller, model and template to use "article" prefix

Further refactor would be deleting controller, however it's not that simple as it's dependent on other controllers (Application, Article).

### CODE REVIEW

File: front/scripts/main/components/ArticleAddPhotoComponent.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+'use strict';
+
+App.ArticleAddPhotoComponent = Em.Component.extend({
+	classNames: ['addphoto-component'],
+
+	actions: {
+		back: function (): void {
+			this.sendAction('back');
+
+			M.track({
```
Comment (vforge) [inline front/scripts/main/components/ArticleAddPhotoComponent.ts]: Please consider creating a function for this tracking as you invoking `M.track` with almost the same data twice.

File: front/scripts/main/components/ArticleAddPhotoComponent.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/// <reference path="../app.ts" />
+'use strict';
+
+App.ArticleAddPhotoComponent = Em.Component.extend({
+	classNames: ['addphoto-component'],
```
Comment (rogatty) [inline front/scripts/main/components/ArticleAddPhotoComponent.ts]: How about `article-add-photo`?

File: front/scripts/main/components/ArticleAddPhotoComponent.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+	actions: {
+		back: function (): void {
```
Comment (rogatty) [inline front/scripts/main/components/ArticleAddPhotoComponent.ts]: `back(): void {`

File: front/scripts/main/router.ts [hunk 4]
```diff
@@ -66,7 +66,7 @@ App.Router.map(function () {
 		path: articlePath + 'edit/:title/:sectionIndex'
 	});
 
-	this.route('addPhoto', { // Symbolic link to AddPhotoController
+	this.route('articleAddPhoto', { // Symbolic link to ArticleAddPhotoController
```
Comment (rogatty) [inline front/scripts/main/router.ts]: I'd remove this comment as it's not really true. Here we specify handler (route) to use on a given path.

File: front/scripts/main/routes/ArticleAddPhotoRoute.ts [hunk 5]
```diff
@@ -4,7 +4,13 @@
...
-App.AddPhotoRoute = Em.Route.extend(App.FullPageMixin, {
+App.ArticleAddPhotoRoute = Em.Route.extend(App.FullPageMixin, {
+	renderTemplate(): void {
+		this.render('article-add-photo', {
+			controller: 'articleAddPhoto'
+		});
+	},
+
 	actions: {
 		error: function (error: any, transition: EmberStates.Transition): boolean {
```
Comment (rogatty) [inline front/scripts/main/routes/ArticleAddPhotoRoute.ts]: `error(error: any, transition: EmberStates.Transition): boolean {`
Convert whole file while you're here.

File: front/scripts/main/routes/ArticleAddPhotoRoute.ts [hunk 6]
```diff
@@ -4,7 +4,13 @@
 
 'use strict';
 
-App.AddPhotoRoute = Em.Route.extend(App.FullPageMixin, {
+App.ArticleAddPhotoRoute = Em.Route.extend(App.FullPageMixin, {
+	renderTemplate(): void {
+		this.render('article-add-photo', {
+			controller: 'articleAddPhoto'
```
Comment (rogatty) [inline front/scripts/main/routes/ArticleAddPhotoRoute.ts]: Would it work without specifying controller?

File: front/scripts/main/components/ArticleAddPhotoComponent.ts (hunk 2 above)
Comment (vforge) [inline front/scripts/main/components/ArticleAddPhotoComponent.ts]: :+1:

File: front/scripts/main/routes/ArticleAddPhotoRoute.ts (hunk 6 above)
Comment (bkoval) [inline front/scripts/main/routes/ArticleAddPhotoRoute.ts]: No :) It would lose model reference.

**Comment by rogatty** [comment]
LGTM

**Comment by vforge** [comment]
LGTM

**Pull request merged**